packages/blocks/src/surface-block/managers/edit-session-manager.ts

```diff
@@ -0,0 +1,317 @@
+import { LineWidth } from '../../_common/types.js';
+import {
+  GET_DEFAULT_LINE_COLOR,
+  GET_DEFAULT_TEXT_COLOR,
+  LineColorsSchema,
+} from '../../page-block/edgeless/components/panel/color-panel.js';
+import {
+  CanvasTextFontFamily,
+  CanvasTextFontStyle,
+  CanvasTextFontWeight,
+  ShapeStyle,
+  StrokeStyle,
+} from '../consts.js';
+import type { EdgelessElementType } from '../edgeless-types.js';
+import {
+  ConnectorEndpointStyle,
+  DEFAULT_FRONT_END_POINT_STYLE,
+  DEFAULT_REAR_END_POINT_STYLE,
+} from '../elements/connector/types.js';
+import { TextAlign, TextVerticalAlign } from '../elements/consts.js';
+import {
+  DEFAULT_SHAPE_FILL_COLOR,
+  DEFAULT_SHAPE_STROKE_COLOR,
+  SHAPE_TEXT_FONT_SIZE,
+  ShapeColorsSchema,
+  ShapeType,
+} from '../elements/shape/consts.js';
+import type { SurfaceService } from '../surface-service.js';
+
+const ConnectorEndpointSchema = z.nativeEnum(ConnectorEndpointStyle);
+const StorkeStyleSchema = z.nativeEnum(StrokeStyle);
+const LineWidthSchema = z.nativeEnum(LineWidth);
+const ShapeStyleSchema = z.nativeEnum(ShapeStyle);
+const ShapeTextFontSizeSchema = z.nativeEnum(SHAPE_TEXT_FONT_SIZE);
+const CanvasTextFontFamilySchema = z.nativeEnum(CanvasTextFontFamily);
+const CanvasTextFontWeightSchema = z.nativeEnum(CanvasTextFontWeight);
+const CanvasTextFontStyleSchema = z.nativeEnum(CanvasTextFontStyle);
+const TextAlignSchema = z.nativeEnum(TextAlign);
+const TextVerticalAlignSchema = z.nativeEnum(TextVerticalAlign);
+const ShapeTypeSchema = z.nativeEnum(ShapeType);
+
+const LastPropsSchema = z.object({
+  connector: z.object({
+    frontEndpointStyle: ConnectorEndpointSchema,
+    rearEndpointStyle: ConnectorEndpointSchema,
+    strokeStyle: StorkeStyleSchema,
+    stroke: LineColorsSchema,
+    strokeWidth: LineWidthSchema,
+    rough: z.boolean(),
+    mode: z.number().optional(),
+  }),
+  brush: z.object({
+    color: LineColorsSchema,
+    lineWidth: LineWidthSchema,
+  }),
+  shape: z.object({
+    shapeType: ShapeTypeSchema,
+    fillColor: ShapeColorsSchema,
+    strokeColor: ShapeColorsSchema,
+    shapeStyle: ShapeStyleSchema,
+    filled: z.boolean().optional(),
+    strokeWidth: z.number().optional(),
+    strokeStyle: StorkeStyleSchema.optional(),
+    color: z.string().optional(),
+    fontSize: ShapeTextFontSizeSchema.optional(),
+    fontFamily: CanvasTextFontFamilySchema.optional(),
+    fontWeight: CanvasTextFontWeightSchema.optional(),
+    fontStyle: CanvasTextFontStyleSchema.optional(),
+    textAlign: TextAlignSchema.optional(),
+    textHorizontalAlign: TextAlignSchema.optional(),
+    textVerticalAlign: TextVerticalAlignSchema.optional(),
+  }),
+  text: z.object({
+    color: z.string(),
+    fontFamily: CanvasTextFontFamilySchema,
+    textAlign: TextAlignSchema,
+    fontWeight: CanvasTextFontWeightSchema,
+    fontStyle: CanvasTextFontStyleSchema,
+    fontSize: z.number(),
+    hasMaxWidth: z.boolean(),
+  }),
+  ['affine:note']: z.object({
```

Comment:
Prefer using `'affine:note'` instead of `['affine:note']` 🤔

Reply:
Okay